### DISCUSSION TITLE
Work with negative SiPM signal

### DISCUSSION THREAD

**Discussion by cozimus** [OpenGammaProject/Open-Gamma-Detector#56] [Q&A]
First of all, thank you very much for your effort in this project, it's just an amazing combination of programming, electronics and physics! I tried building my detector, but I think I faced an issue... I have some electronic embeddd to my SiPMs that gives a negative-shaped waveform, and I think this is not handled correctly. There is any kind of workaround for that?
Thanks a lot 🙏

**Comment by NuclearPhoenixx** [comment]
Hey there, thanks for the kind words! 😄 

Unfortunately, the SiPM pulse input to the device only accepts positive voltages since they go directly into the preamp that is using a single +3.3V supply rail. Everything above 3.3V or below 0V will be clamped by an input protection diode, so the same will be true for your negative SiPM pulses.
There is no solution to this on the detector board itself. However, you could put an inverting op amp between your negative SiPM output and the device's SiPM input. That should work just fine if you can get the right op amp.

**Comment by l-a-l** [comment]
我使用了一个反相的前置放大器后面用电阻衰减到3v左右通到这个能谱仪可以工作。
